app/src/main/java/io/gnosis/safe/ui/transactions/details/view/TxSettingsActionView.kt

```diff
@@ -0,0 +1,127 @@
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+    init {
+        orientation = LinearLayout.VERTICAL
+        gravity = Gravity.LEFT
+    }
+
+    private val binding by lazy { ViewTxActionBinding.inflate(LayoutInflater.from(context), this) }
+    private fun clear() {
+        removeAllViews()
+    }
+
+    fun setActionInfo(txInfo: TransactionInfo) {
+        clear()
+        val settingsChange = txInfo as TransactionInfo.SettingsChange
+        addStringItem(settingsMethodTitle.get(settingsChange.dataDecoded.method)!!)
+
+        val params = settingsChange.dataDecoded.parameters
+        when (settingsChange.dataDecoded.method) {
+            METHOD_CHANGE_MASTER_COPY -> {
+                val mainCopy = params.getValueByName("_masterCopy")?.asEthereumAddress()!!
+                val label = mainCopy.getVersionForAddress()
+                addLabeledAddressItem(mainCopy, label)
+            }
+            METHOD_CHANGE_THRESHOLD -> addStringItem(params.getValueByName("_threshold")!!)
+            METHOD_ADD_OWNER_WITH_THRESHOLD -> {
+                addAddressItem(params.getValueByName("owner")?.asEthereumAddress()!!)
+                addStringItem(settingsMethodTitle[METHOD_CHANGE_THRESHOLD]!!)
+                addStringItem(params.getValueByName("_threshold")!!)
+            }
+
+            METHOD_REMOVE_OWNER -> {
+                addAddressItem(params.getValueByName("owner")?.asEthereumAddress()!!)
+                addStringItem(settingsMethodTitle[METHOD_CHANGE_THRESHOLD]!!)
+                addStringItem(params.getValueByName("_threshold")!!)
+            }
+            METHOD_SET_FALLBACK_HANDLER -> {
+                val label =
+                    if (DEFAULT_FALLBACK_HANDLER == params.getValueByName("handler")?.asEthereumAddress()) {
+                        "DefaultFallBackHandler"
+                    } else {
+                        "Unknown"
+                    }
+                addLabeledAddressItem(params.getValueByName("handler")?.asEthereumAddress()!!, label)
+            }
+            METHOD_SWAP_OWNER -> {
+                addAddressItem(params.getValueByName("oldOwner")?.asEthereumAddress()!!)
+                addStringItem(settingsMethodTitle[METHOD_ADD_OWNER_WITH_THRESHOLD]!!)
+                addAddressItem(params.getValueByName("newOwner")?.asEthereumAddress()!!)
+            }
+            METHOD_ENABLE_MODULE, METHOD_DISABLE_MODULE -> {
+                addAddressItem(params.getValueByName("module")?.asEthereumAddress()!!)
+            }
+        }
+    }
+
+    private fun addStringItem(text: String) {
+        val addressItem = TextView(context)
+        val layoutParams = LinearLayout.LayoutParams(LinearLayout.LayoutParams.MATCH_PARENT, dpToPx(TxTransferActionView.ADDRESS_ITEM_HEIGHT))
+        layoutParams.setMargins(dpToPx(TxTransferActionView.ADDRESS_ITEM_MARGIN_LEFT), 0, 0, dpToPx(TxTransferActionView.MARGIN_VERTICAL))
```

Comment:
either there should be a global margin for all items inside this view, or string item should have it's own. Using margin from a different view doesn't look good. You could define margin inside of the companion object or have then in dimens.xml